runtime/gc_glue_java/HeapRegionDescriptorStandardExtension.hpp

```diff
@@ -69,24 +69,81 @@ class MM_HeapRegionDescriptorStandardExtension : public MM_BaseNonVirtual
 	bool
 	initialize(MM_EnvironmentBase *env)
 	{
-		if (NULL == (_unfinalizedObjectLists = MM_UnfinalizedObjectList::newInstanceArray(env, _maxListIndex))) {
-			return false;
+		if ((NULL == (_unfinalizedObjectLists = MM_UnfinalizedObjectList::newInstanceArray(env, _maxListIndex, NULL, 0)))
+			|| (NULL == (_ownableSynchronizerObjectLists = MM_OwnableSynchronizerObjectList::newInstanceArray(env, _maxListIndex, NULL, 0)))
+			|| (NULL == (_continuationObjectLists = MM_ContinuationObjectList::newInstanceArray(env, _maxListIndex, NULL, 0)))
+			|| (NULL == (_referenceObjectLists = MM_ReferenceObjectList::newInstanceArray(env, _maxListIndex, NULL, 0)))
+		) {
+			goto failed;
 		}
 
-		if (NULL == (_ownableSynchronizerObjectLists = MM_OwnableSynchronizerObjectList::newInstanceArray(env, _maxListIndex))) {
-			return false;
+		return true;
+
+failed:
+	releaseLists(env);
```

Comment:
incorrect indentation
beside, I'm not sure there is much value in 'goto' - you could've just call release and return false from there
but we do have this pattern here and there - up to you what you like more

Reply:
Simplified and removed goto